include/envoy/request_info/request_info.h

```diff
@@ -71,63 +71,123 @@ class RequestInfo {
   /**
    * @return the time that the first byte of the request was received.
    */
-  virtual SystemTime startTime() const PURE;
+  virtual const SystemTime& startTime() const PURE;
```

Comment:
If we leave this as is, maybe change this to firstDownstreamRxByteReceived for consistency?

If it's returning an absolute time where everything else does duration you can leave it "special" :-)

Reply:
I'll leave it "special" and convert all the other methods to return nanos instead. I think that gives us the most bang for our buck.